Title       : x11-base/xorg-server-1.3.0.0 uses wrong screen resolutions on a dual-head systems
Assignee    : x11@gentoo.org
Reported    : 2007-05-23 16:53 0000
Updated     : 2007-10-02 19:39:28 0000
Status      : NEW
Severity    : normal
Priority    : P2
Reporter    : jouni.rinne@pp1.inet.fi
Product     : Gentoo Linux
Component   : Server
Keywords    : Inclusion
CC          : bas@lijer.nl
CC          : daniel.pielmeier@googlemail.com
CC          : frlinux@frlinux.net
CC          : ml81@gmx.de
Comments    : 14
Attachments : 1
URL         : http://bugs.gentoo.org/show_bug.cgi?id=179545

commit 9c80eda826448822328bb678a7d284cc43fffb17
Author: Adam Jackson <ajax@benzedrine.nwnk.net>
Date:   Wed Apr 25 16:35:04 2007 -0400

    Disable RANDR's fake Xinerama protocol when there's more than one screen.

    ... in the protocol sense.  Xinerama doesn't have any provision for more
    than one protocol screen each with its own geometry.

    Red Hat bug #231257.

diff --git a/randr/rrxinerama.c b/randr/rrxinerama.c
index 1db27f1..2a57e4e 100644
--- a/randr/rrxinerama.c
+++ b/randr/rrxinerama.c
@@ -428,6 +428,14 @@ RRXineramaExtensionInit(void)
 	return;
 #endif
 
+    /*
+     * Xinerama isn't capable enough to have multiple protocol screens each
+     * with their own output geometry.  So if there's more than one protocol
+     * screen, just don't even try.
+     */
+    if (screenInfo.numScreens > 1)
+	return;
+
     (void) AddExtension(PANORAMIX_PROTOCOL_NAME, 0,0,
 			ProcRRXineramaDispatch,
 			SProcRRXineramaDispatch,
